Source/Extensions/Blazorise.DataGrid/DataGrid.razor.cs

```diff
@@ -570,7 +570,8 @@ public Task ClearFilter()
         public void FilterData()
         {
             FilterData( Data?.AsQueryable() );
-            StateHasChanged();
+
+            InvokeAsync( StateHasChanged );

```

Comment:
I might be wrong... this is only needed for when you need to synchronize with the "UI" thread. And that's why I did not use it. Since the methods are already synchronous I just used the synchronous version. Eitherway it doesn't hurt I guess.

Reply:
From my experience, it is best to always wrap it in InvokeAsync. Otherwise, some nasty things might happen.